### PR TITLE
BW-758 #comment changes for question of day component unit test

### PR DESCRIPTION
**Pull request by milansar**

### CODE REVIEW

File: projects/trivia/src/app/dashboard/component/question/question.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+                            answers[3] = tempAnswers[0];
+                            answers[2] = tempAnswers[2];
+                            answers[1] = tempAnswers[3];
+                            answers[0] = tempAnswers[1];
+                          }
+                          return answers;
+                    }
+                }},
+                provideMockStore( {
+                    selectors: [
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/question/question.component.spec.ts:44]: please initial value     initialState: {},

File: projects/trivia/src/app/dashboard/component/question/question.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+    it('Verify Correct answer', () => {
+        mockStore.overrideSelector<AppState, Partial<CoreState>>(appState.coreState, {
+            categories: TEST_DATA.categoryList,
+            questionOfTheDay: TEST_DATA.questionOfTheDay
+          });
+        mockStore.refreshState();
+        fixture.detectChanges();
+
+        expect(component.correctAnswerText).toEqual('Object Class');
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/question/question.component.spec.ts:130]: it appears that we are checking serialized text of string class, we should check actual text provided in questionOftheDay or correctAnswerText object if it is object

File: projects/trivia/src/app/dashboard/component/question/question.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+      fixture.detectChanges();
+
+      component.getNextQuestion();
+
+      expect(spy).toHaveBeenCalledWith(
+          new QuestionActions().getQuestionOfTheDay()
+        );
+    });
+
+    it('verify getNextQuestion function works', async () => {
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/question/question.component.spec.ts:202]: it  appears to be duplicate of test case 'verify get Next Question action dispatches correctly'

File: projects/trivia/src/app/dashboard/component/question/question.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        });
+      mockStore.refreshState();
+      fixture.detectChanges();
+      component.answerButtonClicked(TEST_DATA.questionOfTheDay.answers[2]);
+      expect(spy).toHaveBeenCalledWith(
+        new GameActions().UpdateQuestionStat(TEST_DATA.questionOfTheDay.id, 'CORRECT')
+      );
+    });
+
+    it('verify if the answer is correct UpdateQuestionStat action should fire with wrong argument', async () => {
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/question/question.component.spec.ts:259]: i think here in test case description you meant verify if the is wrong........

File: projects/trivia/src/app/dashboard/component/question/question.component.spec.ts (hunk 2 above)
Comment (piya99) [reply projects/trivia/src/app/dashboard/component/question/question.component.spec.ts:130]: In this scenario I am checking current answer text, which is string type

**Comment by milansar** [comment]
@anihalaney , PR is ready for your review

**Pull request merged**